pkg/util/util.go

```diff
@@ -785,11 +793,46 @@ func GetGitHubZipURL(repoURL string) (string, error) {
 		repo = strings.TrimSuffix(repo, ".git")
 	}
 
-	// TODO: pass branch or tag from devfile
-	branch := "master"
+	// Extract reference checkout for project
+	references := make(map[string]string)
+	if project.Source.Branch != nil && *project.Source.Branch != "" {
+		references[githubBranchKey] = *project.Source.Branch
+	}
+
+	if project.Source.CommitId != nil && *project.Source.CommitId != "" {
+		references[githubCommitIDKey] = *project.Source.CommitId
+	}
+
+	if project.Source.Tag != nil && *project.Source.Tag != "" {
+		references[githubTagKey] = *project.Source.Tag
+	}
+
+	if project.Source.StartPoint != nil && *project.Source.StartPoint != "" {
+		references[githubStartPointKey] = *project.Source.StartPoint
+	}
+
+	if len(references) > 1 {
```

Comment:
I don't believe this is a good way of checking to see if there are any references? See below comment with regards to using a struct instead.

Reply:
> I don't believe this is a good way of checking to see if there are any references?

+1

Reply:
@prietyc123 Any thoughts on: https://github.com/openshift/odo/pull/3100#discussion_r423143935 ?

Reply:
> @prietyc123 Any thoughts on: #3100 (comment) ?

https://github.com/openshift/odo/pull/3100#discussion_r428086370